fix: Overview tab infinite loading for BCDC-only countries (Russia, Brazil, Japan) (5348)

Narek13 opened the pull request:
## Issue Description
The Overview tab displays an infinite loading spinner for BCDC-only countries (Brazil, Japan, and Russia). The tab never finishes loading because `featuresIsReady` never becomes `true` for these countries, as they don't have APM (Alternative Payment Methods) feature support.

**Affected Countries:**
- Brazil
- Japan  
- Russia

**Symptoms:**
- Overview tab shows perpetual loading animation
- Tab becomes functional immediately when store country is changed to US, France, Germany, or other countries with APM support
- Issue occurs in both white-label and branded modes

## PR Description
Fixes infinite loading on Overview tab for BCDC-only countries by implementing a more flexible features readiness check.

**Root Cause:**
For Brazil, Japan, and Russia, the `featuresIsReady` flag never becomes `true` because these countries don't have APM features available. The Overview tab was waiting for this flag before rendering, causing an infinite loading state.

**Solution:**
Fixed the root cause in `useFeatures()` hook by removing the `featuresData.length > 0` condition that prevented `isReady` from being set to `true` when the features array is empty. The hook now properly sets `isReady` to `true` after a successful API fetch regardless of whether features are available, as an empty array is a valid response for countries without APM support (Brazil, Japan, Russia).

This ensures that any component using `FeaturesHooks.useFeatures()` will work correctly for all countries without requiring individual workarounds.

**Impact:**
- ✅ **BCDC-only countries**: Overview tab now loads properly for Brazil, Japan, and Russia
- ✅ **Countries with APM**: No change in behavior, continues working as before
- ✅ **User experience**: Merchants can access Overview tab regardless of store country
- ✅ **No breaking changes**: Gracefully handles both scenarios (with and without features)


<table>
<tr>
<td width="50%">

**Before**

</td>
<td width="50%">

**After**

</td>
</tr>

<tr>
<td width="50%">

<img width="1015" height="647" alt="Screenshot 2025-10-15 at 18 08 14" src="https://github.com/user-attachments/assets/5eadb8d2-7a04-438a-a7e3-20a6dcc2dad7" />

</td>
<td width="50%">

<img width="1037" height="515" alt="Screenshot 2025-10-15 at 18 08 24" src="https://github.com/user-attachments/assets/4c979ac8-ab60-4a87-8e27-07fd9c70f050" />

</td>
</tr>
</table>

Narek13 committed:
fix: Overview tab infinite loading for BCDC-only countries

Fix Overview tab loading indefinitely for Brazil, Japan, and Russia
by treating features as ready when features array is empty. These
BCDC-only countries don't have APM feature support, causing
featuresIsReady to never become true and blocking Overview tab render.

Introduces effectiveFeaturesReady check that considers features ready
either when featuresIsReady is true or when features array is empty,
allowing Overview tab to load properly for all country types.

puntope commented on modules/ppcp-settings/resources/js/Components/Screens/Settings/Tabs/TabOverview.js:
```diff
@@ -8,12 +8,16 @@ import usePaymentGatewaySync from '../../../../hooks/usePaymentGatewaySync';
-	if ( ! areTodosReady || ! merchantIsReady || ! featuresIsReady ) {
+	// For countries without features (Brazil, Japan, Russia), consider features ready if array is empty
+	const effectiveFeaturesReady =
+		featuresIsReady || ( features && features.length === 0 );
```
It would be more efficient to fix the `isReady` logic in `FeaturesHooks.useFeatures()` instead. So, in case we use  `FeaturesHooks.useFeatures()` somewhere else, now or in the future, we won't need to add the patch again.

puntope replied:
Mostly the issue comes from https://github.com/woocommerce/woocommerce-paypal-payments/blob/develop/modules/ppcp-settings/resources/js/data/features/hooks.js#L35

However, all that useEffect duplicated logic could be totally removed if we fix the resolver

https://github.com/woocommerce/woocommerce-paypal-payments/blob/develop/modules/ppcp-settings/resources/js/data/features/hooks.js#L35

Basically, this should be` response?.data?.features`

Narek13 replied:
Yes, thanks @puntope for the suggestion. I have changed the logic here: 1886ea6c7d10979fe3d05121c256bdcf0f2bce85

Narek13 committed:
fix: always set features ready state after successful API fetch

Remove featuresData.length check that prevented isReady from becoming
true when features array is empty. Empty array is a valid response for
countries without APM support (Brazil, Japan, Russia).

Fixes infinite loading in Overview tab and other components depending
on featuresIsReady for BCDC-only countries.

puntope commented on modules/ppcp-settings/resources/js/data/features/hooks.js:
```diff
@@ -53,12 +53,11 @@ export const useFeatures = () => {
-					return { success: true, features: featuresData };
-				}
-				return { success: false, features: [] };
+				// Set features and ready regardless of array length
```
I would omit redundant comments.

puntope commented on modules/ppcp-settings/resources/js/data/features/hooks.js:
```diff
@@ -32,10 +32,10 @@ export const useFeatures = () => {
+					// Set features regardless of length
+					await setFeatures( featuresData );
+					// Always set ready after successful fetch
+					await setIsReady( true );
```
I would omit redundant comments

Narek13 committed:
chore: remove redundant inline comments
